### PR TITLE
OSD-29156: Added test cases for pool.go

### PR DESCRIPTION
**Pull request by drakshakhan**
This PR adds test cases for the pool.go in Aao package, improving the test coverage from 0% to 62.8%. The following changes were made:

Test Case Addition
Added unit tests to thoroughly test the functionality in pool.go.

Test Frameworks Used
Testify for assertions and test suite management.
Assert for more expressive checks and validations.

Coverage Details
The code coverage for the pool.go file has been increased to 68.2%. This includes tests for various scenarios, such as edge cases and failure paths.

Test Execution
Tests were executed using the following commands:
go test -coverprofile=coverage.out ./... to run the tests and generate coverage information.
go tool cover -html=coverage.out to visualize the coverage and ensure completeness.

Verification
All tests pass successfully, confirming that the added tests work as expected and improve the overall test coverage.

Jira Story - https://issues.redhat.com/browse/OSD-29156

### CODE REVIEW

**Comment by drakshakhan** [comment]
@clcollins, @dustman9000
Just a reminder. Please approve if no other comments.
CC - @rpodishe

File: cmd/aao/pool_test.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	client.Client
+	mock.Mock
+}
+
+func (m *MockKubeClient) List(ctx context.Context, obj client.ObjectList, opts ...client.ListOption) error {
+	args := m.Called(ctx, obj, opts)
+	return args.Error(0)
+}
+
+func TestRun(t *testing.T) {
```
Comment (petrkotas) [inline cmd/aao/pool_test.go:25]: Would you please also add the validation of the print output? you can look at the PR https://github.com/openshift/osdctl/pull/695 that is already doing it.

**Review by petrkotas — CHANGES_REQUESTED** [review]
This is a good work. I have one nit, and than all good.

File: go.mod [hunk 2]
```diff
@@ -9,9 +9,9 @@ require (
 	github.com/Dynatrace/dynatrace-operator v0.14.2
 	github.com/PagerDuty/go-pagerduty v1.8.0
 	github.com/andygrunwald/go-jira v1.16.0
-	github.com/aws/aws-sdk-go-v2 v1.36.1
```
[the remark this answers is not in the document]
Comment (drakshakhan) [reply go.mod]: Hi @petrkotas , I have removed the version upgrades.

File: cmd/aao/pool_test.go (hunk 1 above)
Comment (drakshakhan) [reply cmd/aao/pool_test.go:23]: Hi @petrkotas , I have added the validation of the print output in the same way i did in the PR #695. Could you please take a look?

**Review by joshbranham — COMMENTED** [review]
/lgtm
